### PR TITLE
Viite 1916 check error in road network error

### PR DESCRIPTION
**Pull request by Campanha**

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala [hunk 1]
```diff
@@ -119,16 +120,21 @@ class RoadNetworkService {
...
       }
     }
 
       withDynTransaction {
         try {
           val roadsInChunk = roadwayDAO.fetchAllByRoadNumbers(options.roadNumbers)
-          val linearLocationsInChunk = linearLocationDAO.fetchByRoadways(roadsInChunk.map(_.roadwayNumber).distinct.toSet).groupBy(_.roadwayNumber)
-          val roadways = roadsInChunk.groupBy(g => (g.roadNumber, g.roadPartNumber))
-
+          val distinctDateRoads = roadsInChunk.groupBy(_.roadPartNumber).flatMap{ p =>
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala]: Shouldn't here be grouping by road number too - like before?

**Comment by skosonen** [comment]
Removed duplicate images.

**Comment by skosonen** [comment]
When I run the checker, I get this error:
ORA-01791: not a SELECTed expression

**Comment by skosonen** [comment]
Tests pass on Jenkins: #491

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala [hunk 2]
```diff
@@ -52,65 +53,108 @@ class RoadNetworkService {
...
-            (middleCalibrationPointsError ++ edgeCalibrationPointsError).map { loc =>
-              RoadNetworkError(0, roadways.find(_.roadwayNumber == loc.roadwayNumber).get.id, loc.id, AddressError.InconsistentContinuityCalibrationPoints, System.currentTimeMillis(), options.currNetworkVersion)
+            ).map { loc =>
+              val optMidRoadway = roadways.find(_.roadwayNumber == loc.roadwayNumber)
+              RoadNetworkError(0, if (optMidRoadway.nonEmpty) optMidRoadway.get.id else 0L, loc.id, AddressError.InconsistentContinuityCalibrationPoints, System.currentTimeMillis(), options.currNetworkVersion)
             }
-          }
+
+          startLocationErrors ++ middleCalibrationPointsErrors ++ endLocationErrors
+          //          val firstOpt = allLocations.find(loc => loc.orderNumber == 1 && loc.calibrationPoints._1.nonEmpty && loc.calibrationPoints._1.get == 0)
```
Comment (hestrompa) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala]: Remove the commented code please

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala [hunk 3]
```diff
@@ -119,79 +163,82 @@ class RoadNetworkService {
...
-          case e: Exception =>
-            logger.error(e.getMessage)
-            dynamicSession.rollback()
         }
+
+      } catch {
+        case e: SQLIntegrityConstraintViolationException => logger.error("A road network check is already running")
+        case e: SQLException =>
+          logger.info("SQL Exception")
+          println(s"\n" + e.getMessage + s"\n"+ e.printStackTrace)
```
Comment (hestrompa) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala]: Do we need to keep the printStackTrace?

**Pull request merged**